game/scripts/vscripts/units/stopfightingyourself.lua

```diff
@@ -0,0 +1,194 @@
+require('libraries/garbagecollection')
+
+-- Entry Function
+function Spawn(entityKeyValues)
+  local disabled = false
+  if disabled then
+    return
+  end
+
+  print("Starting AI for " .. thisEntity:GetUnitName() .. " " .. thisEntity:GetEntityIndex())
+  thisEntity:SetContextThink('StopFightingYourselfThink', partial(Think, thisEntity), 1)
+
+  ABILITY_dupe_heroes = thisEntity:FindAbilityByName('boss_stopfightingyourself_dupe_heroes')
+
+  thisEntity.GC = GarbageCollection:Add(thisEntity.illusions, 10)
+end
+
+function Think(state, target)
+  -- NOTE: I'm thinking too long
+  -- BUG: Not attacking
+  -- BUG: illusions do not get deleted correctly
```

Comment:
I already set a cap at 32. And technically I'm not using illusions but heroes of team `DOTA_TEAM_NEUTRALS`.